### PR TITLE
Add additional testcases for batch uploads

### PR DESCRIPTION
**Pull request by rfk**
@rtilder here's a small contribution to the cause - some additional testcases covering more corner-case aspects of the batch upload behaviour.  Also a hopefuly fix for the no-default-values-on-text-columns problem.

### CODE REVIEW

File: syncstorage/tests/functional/test_storage.py [hunk 1]
```diff
@@ -1378,6 +1378,140 @@ def test_we_dont_need_no_stinkin_batches(self):
...
+        # The payloads should be unchanged
+        resp = self.app.get(collection + '?full=1')
+        res = resp.json
+        res.sort(key=lambda bso: bso['id'])
+        self.assertEquals(len(res), 3)
+        self.assertEquals(res[0]['payload'], 'ayy')
+        self.assertEquals(res[1]['payload'], 'bea')
+        self.assertEquals(res[2]['payload'], 'see')
+
+        # If we wait, the ttls should kick in
```
Comment (rfk) [inline syncstorage/tests/functional/test_storage.py:1486]: Unresolved semantic issue: should the ttl time count from the time at which the item is posted, or the time at which the batch is committed?  It probably doesn't matter, but perhaps we should pick one and be explicit about it.

**Comment by rfk** [comment]
(btw these tests all currently fail, hopefully they can point us at tweaks to be made to the db queries)

File: syncstorage/tests/functional/test_storage.py (hunk 1 above)
Comment (rnewman) [inline syncstorage/tests/functional/test_storage.py:1486]: No strong preference; these things are on the order of weeks or months. But let's say committed.

Comment (rfk) [inline syncstorage/tests/functional/test_storage.py:1486]: Agreed; I added an explicit test case to assert this
